engine/class_modules/paladin/sc_paladin_protection.cpp

```diff
@@ -522,6 +537,20 @@ struct shield_of_the_righteous_t : public holy_power_consumer_t<paladin_melee_at
       );
     }
 
+    if ( !background )
```

Comment:
Could probably simplify this to avoid repeating a lot of similar code.
Maybe something like
```
    {
      if ( p() -> buffs.shining_light_free -> up() || p() -> buffs.shining_light_stacks -> at_max_stacks() )
      {
        p() -> buffs.shining_light_stacks -> expire();
        p() -> buffs.shining_light_free -> trigger();
      }
      else
        p() -> buffs.shining_light_stacks -> trigger();
    }
```